plugins/discovery-gce/src/test/java/org/elasticsearch/discovery/gce/GceMetadataServiceMock.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.discovery.gce;
+
+import com.google.api.client.http.HttpTransport;
+import org.elasticsearch.cloud.gce.GceComputeServiceImpl;
+import org.elasticsearch.cloud.gce.GceMetadataServiceImpl;
+import org.elasticsearch.common.network.NetworkService;
+import org.elasticsearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+/**
+ * Mock for GCE Metadata Service
+ */
+public class GceMetadataServiceMock extends GceMetadataServiceImpl {
```

Comment:
++ for this setup, but it makes the GceMetadataService interface redundant, right? we can just call GceMetadataServiceImpl  GceMetadataService and remove the interface, right ?